src/Features/Core/Shared/Extensions/DiagnosticAnalyzerExtensions.cs

```diff
@@ -27,7 +28,7 @@ public static DiagnosticAnalyzerCategory GetDiagnosticAnalyzerCategory(this Diag
                 // to be able to operate on a limited span of the document. In practical terms, no analyzer
                 // can have both SemanticDocumentAnalysis and SemanticSpanAnalysis as categories.
                 bool cantSupportSemanticSpanAnalysis = false;
-                var analyzerActions = driver.GetAnalyzerActionsAsync(analyzer).WaitAndGetResult(driver.CancellationToken);
+                var analyzerActions = await driver.GetAnalyzerActionsAsync(analyzer).ConfigureAwait(false);
```

Comment:
I don't see any correctness issues. I'm sure I'm being dense, but it's not obvious to me why these changes make things better. Does less total work end up getting done, or is it important to avoid blocking at a key point, or something else?